src/Compilers/CSharp/Portable/Symbols/Metadata/PE/TupleTypeDecoder.cs

```diff
@@ -200,6 +204,52 @@ private TypeSymbol DecodeType(TypeSymbol type)
             }
         }
 
+        private PointerTypeSymbol DecodePointerType(PointerTypeSymbol type)
+        {
+            return type.WithPointedAtType(DecodeTypeInternal(type.PointedAtTypeWithAnnotations));
+        }
+
+        private FunctionPointerTypeSymbol DecodeFunctionPointerType(FunctionPointerTypeSymbol type)
+        {
+            var parameterTypes = ImmutableArray<TypeWithAnnotations>.Empty;
+            var paramsModified = false;
+
+            if (type.Signature.ParameterCount > 0)
+            {
+                var paramsBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(type.Signature.ParameterCount);
+
+                for (int i = type.Signature.ParameterCount - 1; i >= 0; i--)
+                {
+                    var param = type.Signature.Parameters[i];
+                    var decodedParam = DecodeTypeInternal(param.TypeWithAnnotations);
+                    paramsModified = paramsModified || !decodedParam.IsSameAs(param.TypeWithAnnotations);
+                    paramsBuilder.Add(decodedParam);
+                }
+
+                if (paramsModified)
+                {
+                    paramsBuilder.ReverseContents();
+                    parameterTypes = paramsBuilder.ToImmutableAndFree();
+                }
+                else
+                {
+                    parameterTypes = type.Signature.ParameterTypesWithAnnotations;
+                    paramsBuilder.Free();
+                }
+            }
+
+            var decodedReturnType = DecodeTypeInternal(type.Signature.ReturnTypeWithAnnotations);
```

Comment:
>var decodedReturnType = DecodeTypeInternal(type.Signature.ReturnTypeWithAnnotations); [](start = 12, length = 85)

This doesn't look right. Return type comes in metadata signature before parameter types. Encoding/decoding should use the same order.

Reply:
Why tests didn't catch the ordering issue.

---
In reply to: [420410164](https://github.com/dotnet/roslyn/pull/43961#discussion_r420410164) [](ancestors = 420410164)

Reply:
They did ;)

---
In reply to: [420413115](https://github.com/dotnet/roslyn/pull/43961#discussion_r420413115) [](ancestors = 420413115,420410164)